File: js/reverseList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    first = true
    ptr = null
    while(head) {
        if (first) {
            ptr = new ListNode(head.val)
            first = false
        }
        else{
            tmp = new ListNode(head.val, ptr)
            ptr = tmp
        }
        head = head.next

    }
    return ptr
};

File: reverse_list.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        node = None
        while head:
            nxt = head.next
            head.next = node
            node = head
            head = nxt
        return node

        # Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        nxt = None
        while head:
            tmp = head.next
            head.next = nxt
            nxt = head
            head = tmp
        return nxt